// ngx-toast-notifier/esm2020/lib/services/ngx-toast.service.mjs
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./ngx-toast-notifier.service";
export class NgxToastService {
    constructor(ngxToastService) {
        this.ngxToastService = ngxToastService;
    }
    onSuccess(title, text) {
        const notification = {
            text,
            title,
            icon: 'success',
            bgColor: this.ngxToastService.defaultConfig.config.bgColors?.success
                ? this.ngxToastService.defaultConfig.config.bgColors.success
                : this.ngxToastService.defaultConfig.default.bgColors.success,
        };
        this.ngxToastService.onShowNotification(notification);
    }
    onInfo(title, text) {
        const notification = {
            text,
            title,
            icon: 'info',
            bgColor: this.ngxToastService.defaultConfig.config.bgColors?.info
                ? this.ngxToastService.defaultConfig.config.bgColors.info
                : this.ngxToastService.defaultConfig.default.bgColors.info,
        };
        this.ngxToastService.onShowNotification(notification);
    }
    onWarning(title, text) {
        const notification = {
            text,
            title,
            icon: 'warning',
            bgColor: this.ngxToastService.defaultConfig.config.bgColors?.warning
                ? this.ngxToastService.defaultConfig.config.bgColors.warning
                : this.ngxToastService.defaultConfig.default.bgColors.warning,
        };
        this.ngxToastService.onShowNotification(notification);
    }
    onDanger(title, text) {
        const notification = {
            text,
            title,
            icon: 'danger',
            bgColor: this.ngxToastService.defaultConfig.config.bgColors?.danger
                ? this.ngxToastService.defaultConfig.config.bgColors.danger
                : this.ngxToastService.defaultConfig.default.bgColors.danger,
        };
        this.ngxToastService.onShowNotification(notification);
    }
}
NgxToastService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.10", ngImport: i0, type: NgxToastService, deps: [{ token: i1.NgxToastNotifierService }], target: i0.ɵɵFactoryTarget.Injectable });
NgxToastService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "13.3.10", ngImport: i0, type: NgxToastService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.10", ngImport: i0, type: NgxToastService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.NgxToastNotifierService }]; } });
