// ngx-toast-notifier/esm2020/lib/services/ngx-toast-notifier.service.mjs
import { Inject, Injectable } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
import { NOTIFICATION_CONFIG, } from '../interfaces/i-notification';
import * as i0 from "@angular/core";
export class NgxToastNotifierService {
    constructor(config) {
        this.$notifications = new BehaviorSubject([]);
        this.$notificationsObs = this.$notifications.asObservable();
        this.defaultConfig = config;
    }
    get notifications() {
        return this.$notifications.value;
    }
    onShowNotification(notification) {
        this.$notifications.next([
            ...this.notifications,
            {
                ...notification,
            },
        ]);
        setTimeout(() => {
            this.onRemove();
        }, this.defaultConfig.config.timeOut ? this.defaultConfig.config.timeOut : this.defaultConfig.default.timeOut);
    }
    onRemove() {
        this.notifications.shift();
        this.$notifications.next(this.notifications);
    }
}
NgxToastNotifierService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.10", ngImport: i0, type: NgxToastNotifierService, deps: [{ token: NOTIFICATION_CONFIG }], target: i0.ɵɵFactoryTarget.Injectable });
NgxToastNotifierService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "13.3.10", ngImport: i0, type: NgxToastNotifierService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.10", ngImport: i0, type: NgxToastNotifierService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [NOTIFICATION_CONFIG]
                }] }]; } });
