// ngx-toast-notifier/esm2020/lib/components/ngx-toast-notifier.component.mjs
import { Component } from '@angular/core';
import { onEnterLeave } from '../animations/animations';
import * as i0 from "@angular/core";
import * as i1 from "../services/ngx-toast-notifier.service";
import * as i2 from "./notification/notification.component";
import * as i3 from "@angular/common";
export class NgxToastNotifierComponent {
    constructor(notificationSvc) {
        this.notificationSvc = notificationSvc;
    }
    ngOnInit() {
        this.$notifications = this.notificationSvc.$notificationsObs;
    }
}
NgxToastNotifierComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.10", ngImport: i0, type: NgxToastNotifierComponent, deps: [{ token: i1.NgxToastNotifierService }], target: i0.ɵɵFactoryTarget.Component });
NgxToastNotifierComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.3.10", type: NgxToastNotifierComponent, selector: "ngx-toast-notifier", ngImport: i0, template: `
    <div class="wrap-notifications">
      <ngx-notification
        [notificationData]="item"
        *ngFor="let item of $notifications | async"
        @onEnterLeave
      >
      </ngx-notification>
    </div>
  `, isInline: true, styles: [":host{display:flex;position:fixed;width:300px;height:auto;max-height:100vh;overflow:auto;top:80px;right:15px;z-index:99}.wrap-notifications{position:relative;width:100%;overflow:hidden}@media only screen and (max-width: 500px){:host{width:95%;top:20px;left:0;right:0;bottom:0;margin:auto}}\n"], components: [{ type: i2.NotificationComponent, selector: "ngx-notification", inputs: ["notificationData"] }], directives: [{ type: i3.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }], pipes: { "async": i3.AsyncPipe }, animations: [onEnterLeave] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.10", ngImport: i0, type: NgxToastNotifierComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'ngx-toast-notifier',
                    template: `
    <div class="wrap-notifications">
      <ngx-notification
        [notificationData]="item"
        *ngFor="let item of $notifications | async"
        @onEnterLeave
      >
      </ngx-notification>
    </div>
  `,
                    styles: [
                        `
      :host {
        display: flex;
        position: fixed;
        width: 300px;
        height: auto;
        max-height: 100vh;
        overflow: auto;
        top: 80px;
        right: 15px;
        z-index: 99;
      }

      .wrap-notifications {
        position: relative;
        width: 100%;
        overflow: hidden;
      }

      @media only screen and (max-width: 500px) {
        :host {
          width: 95%;
          top: 20px;
          left: 0;
          right: 0;
          bottom: 0;
          margin: auto;
        }
      }
    `,
                    ],
                    animations: [onEnterLeave],
                }]
        }], ctorParameters: function () { return [{ type: i1.NgxToastNotifierService }]; } });
