// ngx-toast-notifier/esm2020/lib/ngx-toast-notifier.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { DefaultGlobalConfig, NOTIFICATION_CONFIG } from './interfaces/i-notification';
import { NgxToastNotifierComponent } from './components/ngx-toast-notifier.component';
import { NotificationComponent } from './components/notification/notification.component';
import * as i0 from "@angular/core";
export class NgxToastNotifierModule {
    static forRoot(config = {}) {
        return {
            ngModule: NgxToastNotifierModule,
            providers: [
                {
                    provide: NOTIFICATION_CONFIG,
                    useValue: {
                        default: DefaultGlobalConfig,
                        config
                    }
                }
            ]
        };
    }
}
NgxToastNotifierModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.10", ngImport: i0, type: NgxToastNotifierModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgxToastNotifierModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.3.10", ngImport: i0, type: NgxToastNotifierModule, declarations: [NgxToastNotifierComponent,
        NotificationComponent], imports: [CommonModule], exports: [NgxToastNotifierComponent] });
NgxToastNotifierModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.3.10", ngImport: i0, type: NgxToastNotifierModule, imports: [[
            CommonModule
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.10", ngImport: i0, type: NgxToastNotifierModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        NgxToastNotifierComponent,
                        NotificationComponent
                    ],
                    imports: [
                        CommonModule
                    ],
                    exports: [
                        NgxToastNotifierComponent
                    ]
                }]
        }] });
