// @tgillespie/ngx-youtube-player/esm2020/modules/src/youtube-player-module.mjs
import { NgModule } from "@angular/core";
import { CommonModule } from "@angular/common";
import { YouTubePlayerComponent } from "../../youtube-player/src/youtube-player.component";
import { YouTubeIFrameApiService, YouTubePlayerManagerService } from "../../interfaces/src/services";
import { StandardYouTubePlayerManagerService } from "../../services/src/youtube-player-manager.service";
import { StandardYouTubeIFrameApiService } from "../../services/src/youtube-iframe-api.service";
import * as i0 from "@angular/core";
export class YouTubePlayerModule {
    static forRoot() {
        return {
            ngModule: YouTubePlayerModule,
            providers: [{
                    provide: YouTubePlayerManagerService,
                    useClass: StandardYouTubePlayerManagerService,
                },
                {
                    provide: YouTubeIFrameApiService,
                    useClass: StandardYouTubeIFrameApiService,
                },
            ]
        };
    }
}
YouTubePlayerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.1", ngImport: i0, type: YouTubePlayerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
YouTubePlayerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.1", ngImport: i0, type: YouTubePlayerModule, declarations: [YouTubePlayerComponent], imports: [CommonModule], exports: [YouTubePlayerComponent] });
YouTubePlayerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.1", ngImport: i0, type: YouTubePlayerModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.1", ngImport: i0, type: YouTubePlayerModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [YouTubePlayerComponent],
                    exports: [YouTubePlayerComponent],
                }]
        }] });
