// @tgillespie/ngx-youtube-player/esm2020/services/src/youtube-iframe-api.service.mjs
import { Injectable } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
import { YouTubeIFrameApiService } from "../../interfaces/src/services";
import * as i0 from "@angular/core";
export function getIFrameApi() {
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    return window.YT;
}
export class StandardYouTubeIFrameApiService extends YouTubeIFrameApiService {
    constructor() {
        super();
        this._api = new BehaviorSubject(undefined);
        const maybeExistingApi = getIFrameApi();
        if (!maybeExistingApi) {
            const tag = document.createElement('script');
            tag.src = 'https://www.youtube.com/iframe_api';
            const firstScriptTag = document.getElementsByTagName('script')[0];
            if (!firstScriptTag || !firstScriptTag.parentNode) {
                this._api.error(new Error('Could not find or initialize YouTube IFrame API. No first script tag was found in the document'));
                return;
            }
            firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
            // @ts-ignore
            window.onYouTubeIframeAPIReady = () => {
                this._api.next(getIFrameApi());
            };
        }
        else {
            this._api.next(maybeExistingApi);
        }
    }
    get api() {
        return this._api.value;
    }
    get apiChange() {
        return this._api.asObservable();
    }
}
StandardYouTubeIFrameApiService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.1", ngImport: i0, type: StandardYouTubeIFrameApiService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
StandardYouTubeIFrameApiService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "13.2.1", ngImport: i0, type: StandardYouTubeIFrameApiService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.1", ngImport: i0, type: StandardYouTubeIFrameApiService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return []; } });
