// @tgillespie/ngx-youtube-player/esm2020/services/src/youtube-player-manager.service.mjs
import { Injectable } from '@angular/core';
import { Subject } from 'rxjs';
import { YouTubePlayerManagerService } from "../../interfaces/src/services";
import * as i0 from "@angular/core";
export class StandardYouTubePlayerManagerService extends YouTubePlayerManagerService {
    constructor() {
        super(...arguments);
        this.map = new Map();
        this._newEntry = new Subject();
    }
    get newEntry() {
        return this._newEntry.asObservable();
    }
    add(id, data) {
        if (this.map.has(id)) {
            console.error('Player data has been overwritten for ' + id);
        }
        const rejoinedData = {
            id,
            ...data,
        };
        this._newEntry.next(rejoinedData);
        this.map.set(id, rejoinedData);
    }
    delete(id) {
        this.map.delete(id);
    }
    get(id) {
        return this.map.get(id);
    }
    has(id) {
        return this.map.has(id);
    }
}
StandardYouTubePlayerManagerService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.1", ngImport: i0, type: StandardYouTubePlayerManagerService, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
StandardYouTubePlayerManagerService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "13.2.1", ngImport: i0, type: StandardYouTubePlayerManagerService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.1", ngImport: i0, type: StandardYouTubePlayerManagerService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }] });
